python/lsst/analysis/ap/apdb.py

```diff
@@ -364,7 +364,11 @@ class ApdbSqliteQuery(DbQuery):
 
     def __init__(self, filename, *, butler, instrument, **kwargs):
         super().__init__(butler=butler, instrument=instrument, **kwargs)
-        self._engine = sqlalchemy.create_engine(f"sqlite:///{filename}")
+        # For sqlite, use a larger pool and a faster timeout, to allow many
+        # repeat transactions with the same connection, as transactions on
+        # our sqlite DBs should be small and fast.
+        self._engine = sqlalchemy.create_engine(f"sqlite:///{filename}",
+                                                pool_timeout=5, pool_size=10)
```

Comment:
I guess it only matters for tests that run with many threads, and regular apps are all single-threaded? The `pool_size` should be tied to how many concurrent threads you can run, setting it to a fixed value means there is a chance you can exceed it in some random/rare circumstances. Why not remove the limit entirely?

Reply:
Yes, I think all of our regular applications are single threaded. The error I ran into that made me make this change was probably because a single threaded test was opening connections faster than than the pool could free them.

I'll set it to 200, which should never be exceeded in any kind of normal sqlite usage, but still will prevent a runaway.